src/lib.rs

```diff
@@ -886,4 +1265,98 @@ mod tests {
 
         assert!(vm.create_vcpu(0).is_ok());
     }
+
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_run_code() {
+        use std::io::Write;
+
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+        // This example based on https://lwn.net/Articles/658511/
+        let code = [
+            0xba, 0xf8, 0x03, /* mov $0x3f8, %dx */
+            0x00, 0xd8, /* add %bl, %al */
+            0x04, b'0', /* add $'0', %al */
+            0xee, /* out %al, %dx */
+            0xec, /* in %dx, %al */
+            0xc6, 0x06, 0x00, 0x80, 0x00, /* movl $0, (0x8000) */
+            0x8a, 0x16, 0x00, 0x80, /* movl (0x8000), %dl */
+            0xc6, 0x06, 0x00, 0x20, 0x00, /* movl $0, (0x2000) */
```

Comment:
The comments in the [Firecracker codebase](https://github.com/firecracker-microvm/firecracker/blob/master/kvm/src/lib.rs#L1411) would help.